### ISSUE TITLE
Feature request - gdallocationinfo - reading sample points directly from OGR sources

### ISSUE THREAD

**Issue by Brent-Edwards** [OSGeo/gdal#4069]
As mentioned in the gdallocationinfo examples,[https://gdal.org/programs/gdallocationinfo.html#examples](url) it is possible to read a text file of coordinates and return a new text file of sampled vales. It would be really useful to expand this functionality to support reading sample points from any OGR supported format and potentially writing the sampled values to a new column or data source.

**Comment by msmitherdc** [comment]
Raterstats zonal stats will do this https://pythonhosted.org/rasterstats/manual.html#zonal-statistics

**Comment by Brent-Edwards** [comment]
> Raterstats zonal stats will do this https://pythonhosted.org/rasterstats/manual.html#zonal-statistics

Indeed. There are several tools that provide this functionality. My feature request was related specifically to GDAL/gdallocationinfo.

**Comment by rcoup** [comment]
You can _nearly_ do it with the CSV driver and a pipeline...

```console
$ ogr2ogr -f "CSV" -lco SEPARATOR=SPACE -dialect sqlite -sql 'SELECT X(geometry), Y(geometry) FROM myQueryPoints;' tmp1.csv myQueryPoints.shp
$ tail -n +2 tmp1.csv | gdallocationinfo -geoloc -b 1 -valonly myImage.tif > results.csv
```

Adding support for these would allow dropping the temporary files and doing it in a single pipeline:

1. `-lco HEADERS=NO` in the CSV driver
2. single column files in the CSV driver
3. `-` meaning stdout & stdin in `ogr2ogr`
4. `gdallocationinfo` being able to returning `b1 b2 b3` rather than `b1\nb2\b3` 

(1 & 2 might be solvable via a VRT file?)

**Comment by Brent-Edwards** [comment]
> You can _nearly_ do it with the CSV driver and a pipeline...
> 
> ```
> $ ogr2ogr -f "CSV" -lco SEPARATOR=SPACE -dialect sqlite -sql 'SELECT X(geometry), Y(geometry) FROM myQueryPoints;' tmp1.csv myQueryPoints.shp
> $ tail -n +2 tmp1.csv | gdallocationinfo -geoloc -b 1 -valonly myImage.tif > results.csv
> ```
> 
> Adding support for these would allow dropping the temporary files and doing it in a single pipeline:
> 
> 1. `-lco HEADERS=NO` in the CSV driver
> 2. single column files in the CSV driver
> 3. `-` meaning stdout & stdin in `ogr2ogr`
> 4. `gdallocationinfo` being able to returning `b1 b2 b3` rather than `b1\nb2\b3`
> 
> (1 & 2 might be solvable via a VRT file?)

That's pretty much how I've been doing it - OGR2OGR, drop the header and then feed the headerless text file of coords into gdallocationinfo.

Allowing any OGR supported point data type would also open up the possibility to use spatial indexes to further improve performance. I've found gdallocationinfo to be faster than most other tools into the 10s of thousands of points.
